core/drivers/crypto/caam/caam_jr.c

```diff
@@ -0,0 +1,781 @@
+
+				JR_TRACE("JR id=%d, context @0x%08"PRIxVA"",
+					caller->jobid, (vaddr_t)jobctx);
+				/* Clear the Entry Descriptor DMA */
+				caller->pdesc = 0;
+				caller->jobid = JR_JOB_FREE;
+				found = idx_jr;
+				JR_TRACE("Free space #%d in the callers array",
+					idx_jr);
+			}
+		}
+		cpu_spin_unlock(&jr_privdata->callers_lock);
+
+		/*
+		 * Remove the JR from the output list even if no
+		 * JR caller found
+		 */
+		hal_jr_del_job(jr_privdata->baseaddr);
+
+		/*
+		 * Increment index to next JR output entry taking care that
+		 * is a circular buffer of nbJobs size.
+		 */
+		jr_privdata->outread_index++;
+		jr_privdata->outread_index %= jr_privdata->nbJobs;
+
+		if ((found >= 0) && (jobctx->callbk)) {
+			/* Finally, execute user's callback */
+			jobctx->callbk(jobctx);
+		}
+
+	} while (--nbJobs_done);
+
+	cpu_spin_unlock_xrestore(&jr_privdata->inlock, exceptions);
+
+	return retJobId;
+}
+
+/*
+ * Enqueues a new job in the Job Ring input queue. Keep the caller's
+ * job context in private array.
+ *
+ * Inputs:
+ * jobctx   Caller's job context
+ *
+ * Outputs:
+ * jobId    Job Id enqueued
+ *
+ * Returns:
+ * CAAM_NO_ERROR  Success
+ * CAAM_BUSY      CAAM is busy
+ */
+static enum CAAM_Status do_jr_enqueue(struct jr_jobctx *jobctx, uint32_t *jobId)
+{
+	enum CAAM_Status retstatus = CAAM_BUSY;
+
+	struct caller_info *caller = NULL;
+	uint32_t exceptions = 0;
+	uint32_t job_mask = 0;
+	uint8_t  idx_jr = 0;
+	bool     found = false;
+
+	exceptions = cpu_spin_lock_xsave(&jr_privdata->inlock);
+
+	/* Stay lock until a job is available */
+	/* Check if there is an available JR index in the HW */
+	while (hal_jr_read_nbSlotAvailable(jr_privdata->baseaddr) == 0) {
+		/*
+		 * WFE will be exit be a SEV generated but the
+		 * interrupt handler or by a spin_unlock
+		 */
+		wfe();
+	};
+
+	/*
+	 * There is a space free in the input ring but it doesn't mean
+	 * that the job pushed is completed.
+	 * Completion is out of order. Look for a free space in the
+	 * caller data to push them and get a job id for the completion
+	 *
+	 * Lock the caller information array because dequeue is
+	 * also touching it
+	 */
+	cpu_spin_lock(&jr_privdata->callers_lock);
+	for ((idx_jr = 0), (found = false);
+			(idx_jr < jr_privdata->nbJobs) && (!found); idx_jr++) {
+		if (jr_privdata->callers[idx_jr].jobid == JR_JOB_FREE) {
+			JR_TRACE("Found a space #%d free in the callers array",
+				idx_jr);
+			job_mask = 1 << idx_jr;
+
+			/* Store the caller information for the JR completion */
+			caller = &jr_privdata->callers[idx_jr];
+			caller->jobid  = job_mask;
+			caller->jobctx = jobctx;
+			caller->pdesc  = virt_to_phys((void *)jobctx->desc);
+
+			found = true;
+		}
+	}
+	cpu_spin_unlock(&jr_privdata->callers_lock);
+
+	if (!found) {
+		JR_TRACE("Error didn't find a free space in the callers array");
+		goto end_enqueue;
+	}
+
+	JR_TRACE("Push id=%d, job (0x%08x) context @0x%08"PRIxVA"",
+			jr_privdata->inwrite_index, job_mask, (vaddr_t)jobctx);
```

Comment:
Please align after the `(` above. I think there's some option for checkpatch to complain over this, I'm not sure if it's `-strict` or something else.

Reply:
Problem is when you use tab with 4 or 8 spaces you don't have the same result.
Checkpatch is tab=8 spaces and I'm editing file with tab=4 spaces (because I prefer).
I don't know if there is a checkpatch option for that but for sure checkpatch is complaining if a line cross the 80 characters.

Reply:
optee_os uses the Linux kernel coding style: https://optee.readthedocs.io/general/coding_standards.html.
I should update your file editor to use 8-char tabs when in optee_os source tree, as you probably do when browsing the Linux kernel source tree.